expat/lib/expat.h

```diff
@@ -948,6 +975,16 @@ XMLPARSEAPI(int)
 XML_SetHashSalt(XML_Parser parser,
                 unsigned long hash_salt);
 
+/* Set / get XML parser options, see enum XML_Options
+ *
+ * Added in 2.3.0
+ */
+XMLPARSEAPI(int)
+XML_SetOptions(XML_Parser parser, int options);
+
+XMLPARSEAPI(int)
+XML_GetOptions(XML_Parser parser);
```

Comment:
It would be nice if we could cover those two functions by some simple test(s) as well. Some idea for a test:

1. Call to `XML_GetOptions` returns default options
1. Call to `XML_SetOptions` succeeds
1. Call to `XML_GetOptions` returns what was set by `XML_SetOptions` just before
1. Parsing is started
1. Call to `XML_SetOptions` fails (due to parser started)
1. Call to `XML_GetOptions` still returns unmodified previous options

Could also be split in half to make two tests where the second is dedicated to changes at runtime.

Reply:
Good idea! I added a new test scenario as you proposed.